.github/workflows/publish-dotnet.yaml

```diff
@@ -0,0 +1,136 @@
+---
+name: .NET Publish
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: What version number would you like to use? The version number should be **without** a leading `v`, e.g. `5.7.1` or `6.2.4`.
+        required: true
+
+jobs:
+  # TODO: Consider making this its own workflow to be reused by other workflows.
```

Comment:
Would be great to see this take a more modular shape, similar to the new Unleash release workflow by @gastonfournier.

I assume when we want to publish a new version of Yggdrasil we want to publish all language-specific engines, so we could have a single build action for all OS+arch we're targeting, that when finished calls all the language-specific publish actions.

However this is probably good enough for now and for the scope of this PR.

Reply:
Yeah, for a beta this should be good enough, but for production we'll need orchestration of platform specific builds along with language specific builds. but also the possibility to run separate language builds for bug fixes I imagine

Reply:
> Would be great to see this take a more modular shape
Yes
> However this is probably good enough for now, given the scope of this PR
But also yes.

My idle long term goal was always to do as you suggest but hacky hacky hack hack is the right thing to do now while we iterate on the hard problems. Fan out builds are the the easy problem I think